module/zfs/vdev_queue.c

```diff
@@ -686,7 +709,7 @@ vdev_queue_aggregate(vdev_queue_t *vq, zio_t *zio)
 	    (dio->io_flags & ZIO_FLAG_AGG_INHERIT) == flags &&
 	    (IO_SPAN(first, dio) <= limit ||
 	    (dio->io_flags & ZIO_FLAG_OPTIONAL)) &&
-	    IO_SPAN(first, dio) <= maxblocksize &&
+	    IO_SPAN(first, dio) <= SPA_MAXBLOCKSIZE &&
```

Comment:
Who cares what I/O sizes do we send to the disk as long as it does not affect on-disk format, which it does not.

Reply:
zio_create() asserts only psize <= SPA_MAXBLOCKSIZE.  ABDs don't even have pool reference to assert it.  zio_buf_alloc() called by abd_borrow_buf() also receives only size.  I don't see what else.  May be for small 32bit systems too large memory allocations may be hard, but zfs_vdev_aggregation_limit is already limits it to 1MB, so there should be nothing too insane.  I am trying to avoid spa_maxblocksize() call here, appearing not so light in profiler.